nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -2289,36 +2289,30 @@ private boolean mayBeNullExpr(VisitorState state, ExpressionTree expr) {
                   + " for method invocation "
                   + state.getSourceForNode(expr));
         }
-        // Special case: mayBeNullMethodCall runs handler.onOverrideMayBeNullExpr before dataflow.
-        return mayBeNullMethodCall(state, expr, (Symbol.MethodSymbol) exprSymbol);
+        exprMayBeNull = mayBeNullMethodCall((Symbol.MethodSymbol) exprSymbol);
+        break;
       case CONDITIONAL_EXPRESSION:
       case ASSIGNMENT:
-        exprMayBeNull = nullnessFromDataflow(state, expr);
+        exprMayBeNull = true;
         break;
       default:
         // match switch expressions by comparing strings, so the code compiles on JDK versions < 12
         if (expr.getKind().name().equals("SWITCH_EXPRESSION")) {
-          exprMayBeNull = nullnessFromDataflow(state, expr);
+          exprMayBeNull = true;
         } else {
           throw new RuntimeException(
               "whoops, better handle " + expr.getKind() + " " + state.getSourceForNode(expr));
         }
     }
     exprMayBeNull = handler.onOverrideMayBeNullExpr(this, expr, exprSymbol, state, exprMayBeNull);
-    return exprMayBeNull;
+    return exprMayBeNull && nullnessFromDataflow(state, expr);
```

Comment:
together with the `LibraryModelsHandler` refactoring, this would be the only place where `nullnessFromDataflow` would be getting called still